Frontend: course summary page

mertcan-ozkan opened the pull request:
Course summary page.

nurlandadashov02 commented on app/client/pages/course/courseSummary.js:
```diff
@@ -0,0 +1,220 @@
+
+
+  post.chapters.map((chapter) => {
+    console.log(chapter.chapter_name)
```
remove this please

nurlandadashov02 commented on app/client/pages/course/courseSummary.js:
```diff
@@ -0,0 +1,220 @@
+                &nbsp;
+                &nbsp;
+              </Typography>
+              <Typography variant="h7"  >
+                {post.lecturer.email}
+              </Typography>
```
we should not expose email of teacher

nurlandadashov02 commented on app/client/pages/course/courseSummary.js:
```diff
@@ -0,0 +1,220 @@
+      <Container sx={{ 'margin-bottom': '80px' }}>
+
+        <Box sx={{ overflow: 'hidden', minWidth: 250, mt: 1 }}>
+          <Paper variant='outlined' sx={{ 'margin-bottom': 20 }}>
+            <Grid container spacing={2} sx={{ 'margin-bottom': 20, padding: 5 }}>
+
+              <Grid item sx={{ width: '100%' }}>
+                <Typography variant="h4">
+                  Course Description
+                </Typography>
+
+              </Grid>
+              <Grid item sx={{ width: '100%' }}>
+                <Typography>
+                  {post.course_info}
+
+                </Typography>
+                <Grid item sx={{
+                  width: '100%', display: 'flex',
+                  'flex-wrap': 'wrap'
+                }}>
+                  {post.tags.map((tag) => {
+                    return (
+                      <Paper variant='elevation' sx={{ 'margin-right': 7, 'margin-top': 19 }}>
+                        <Typography sx={{ padding: 1 }}>
+                          {tag}
+                        </Typography>
+                      </Paper>)
+                  })}
+                </Grid>
+
+              </Grid>
+            </Grid>
+          </Paper>
```
Excellent!

nurlandadashov02 commented on app/client/pages/course/courseSummary.js:
```diff
@@ -0,0 +1,220 @@
+                <Table sx={{ minWidth: 250 }} >
+                  <TableHead>
+                    <TableRow>
+                      <TableCell>
+                        &nbsp;
+                        &nbsp;
+                        &nbsp;
+                        &nbsp;
+                        &nbsp;
+                        &nbsp;
+                        &nbsp;
+                        &nbsp;
+                        &nbsp;
+                        &nbsp;
+                        &nbsp;
+                        &nbsp;
+                      </TableCell>
+                      <TableCell>Chapter Name</TableCell>
+                    </TableRow>
+                  </TableHead>
+                  <TableBody>
+                    {post.chapters.map((chapter, index) => {
+                      return (
+                        <TableRow>
+                          <TableCell>{index + 1}</TableCell>
+                          <TableCell>{chapter.chapter_name}</TableCell>
+                        </TableRow>
+                      );
+                    })}
+                  </TableBody>
```
There is no need to have chapter indeces. Just a single columns with chapter names will be fine

nurlandadashov02 commented on app/client/pages/course/courseSummary.js:
```diff
@@ -0,0 +1,220 @@
+
+  return (
+    < >
+      <Grid sx={{ width: '100%', minHeight: 250, 'background-color': '#3D8BE0', height: 250, 'padding-top': 20 }}  >
```
General comment: Use fonts. There will be font for headers and other one for paragraphs. I am determining what fonts will be used with mobile team.

nurlandadashov02 replied:
Another general comment, try to make the page as responsive as possible. So, we would have nice mobile web experience.

nurlandadashov02 commented on app/client/pages/course/courseSummary.js:
```diff
@@ -0,0 +1,220 @@
+
+            <Grid item xs={6} sx={{ width: '60%', 'background-color': '#3178C4' }}  >
+
+              <Button fullWidth variant="contained" sx={{ width: '100%' }}>
+                <Typography variant="h4"  >
+                  Enroll
+                </Typography>
+              </Button>
```
Please reuse Button component here. And change button location

nurlandadashov02 commented on app/client/pages/course/courseSummary.js:
```diff
@@ -0,0 +1,220 @@
+            </Grid>
+            <Grid item xs={6} >
+
+              <Typography variant="h5"  >
+                Rating:  {post.rating}
+              </Typography>
```
do not write "Rating". Put rating value and then stars like in udemy.

4.2 * * * *

nurlandadashov02 commented on app/client/pages/course/courseSummary.js:
```diff
@@ -0,0 +1,220 @@
+    < >
+      <Grid sx={{ width: '100%', minHeight: 250, 'background-color': '#3D8BE0', height: 250, 'padding-top': 20 }}  >
+        <Grid  sx={{ height: '95%', width: '80%', margin: 'auto' }}>
+          < Grid container sx={{ float: 'left', width: '60%', height: '100%', padding: 3, 'grid-row-gap': 20}}>
```
background color should changed to some othe color that will used in other places. Determining colors with mobile team now

nurlandadashov02 commented on app/client/pages/course/courseSummary.js:
```diff
@@ -0,0 +1,220 @@
+        <Grid  sx={{ height: '95%', width: '80%', margin: 'auto' }}>
+          < Grid container sx={{ float: 'left', width: '60%', height: '100%', padding: 3, 'grid-row-gap': 20}}>
+            <Grid item xs={12} sx={{ margin: 'auto' }}>
+              <Typography variant="h3"   >
+                {post.title}
+              </Typography>
```
General comment for text: Try to create contrast between the text and background color. For instance, the description and chapters are very nice since black has high contrast with white. However, page header text is black and it does not have high contrast with blue background.

nurlandadashov02 commented on app/client/pages/course/courseSummary.js:
```diff
@@ -0,0 +1,220 @@
+
+
+
+const post = {
```
for future: path should not be -> `/course/courseSummary.js`, it should be `/course/[course_id].js`
Then you can get course_id:
```
const router = useRouter();
const { course_id } = router.query;
```
You can then make request to server to fetch data for the page.

nurlandadashov02 reviewed (changes requested):
Overall, you did a great job Mertcan. There are some minor issues that need to resolved. Thanks!

mertcan-ozkan commented:
following changes are made:
-rating is shown with stars now as nurlan requested. (used berke's css by in this. Thanks berke :) )
-moved some inline styles to a different file
-using fonts now, instead of mui typography
-removed the email part 
-removed chapter indeces
-removed unnecessary code like console.log statements
-changed button location

nurlandadashov02 reviewed (approved):
Thanks Mertcan.
